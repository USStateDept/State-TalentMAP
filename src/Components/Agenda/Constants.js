
/* eslint-disable import/prefer-default-export */
export const pillColors = {
  Withdrawn: '#227c9dff', // dark blue
  Disapproved: '#17c3b2ff', // turquoise
  Approved: '#2d6e0eff', // dark green
  Deferred: '#E08A00', // orange
  'Deferred - Proposed Position': '#E08A00', // orange
  Ready: '#990000', // dark red
  Held: '#6421a2ff', // purple
  'Not Ready': '#BA70FF', // violet
  'Out of Order': '#b38600', // dark orange
  PIP: '#000080', // navy
  'Move to ML/ID': '#ed2038ff', // red
  Default: '#513C2C', // brown
};
/* eslint-enable import/prefer-default-export */
