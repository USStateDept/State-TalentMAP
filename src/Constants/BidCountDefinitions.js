const BID_COUNT_DEFINITIONS = {
  totalBids: { title: 'Total bids', definition: 'Total number of bids on this position.' },
  atGradeBids: { title: 'At-Grade bids', definition: 'Number of bidders who are at-Grade.' },
  inSkillBids: { title: 'In-Skill bids', definition: 'Number of bidders in-Skill.' },
  atGradeInSkillBids: { title: 'At-Grade and in-Skill bids', definition: 'Number of bidders at-Grade and in-Skill.' },
};

export default BID_COUNT_DEFINITIONS;
