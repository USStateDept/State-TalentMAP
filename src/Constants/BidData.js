// prop names from the API
export const APPROVED_PROP = 'approved';
export const CLOSED_PROP = 'closed';
export const DRAFT_PROP = 'draft';
export const DECLINED_PROP = 'declined';
export const HAND_SHAKE_ACCEPTED_PROP = 'handshake_accepted';
export const HAND_SHAKE_DECLINED_PROP = 'handshake_declined';
export const PRE_PANEL_PROP = 'pre_panel';
export const IN_PANEL_PROP = 'in_panel';
export const SUBMITTED_PROP = 'submitted';
export const PANEL_RESCHEDULED_PROP = 'panel_rescheduled';
export const HAND_SHAKE_NEEDS_REGISTER_PROP = 'handshake_needs_registered';

// tmap-specific props
export const HAND_SHAKE_OFFERED_PROP = 'handshake_offered';
export const HAND_SHAKE_OFFER_ACCEPTED_PROP = 'handshake_accepted';
export const HAND_SHAKE_OFFER_DECLINED_PROP = 'handshake_declined';
export const HAND_SHAKE_REVOKED_PROP = 'handshake_offer_revoked';

export const BID_EXPLANATION_TEXT = {
  [SUBMITTED_PROP]: {
    text: "The Bureau is reviewing candidates' qualifications.",
    link: "Learn more about the Bureau's role.",
    term: 'Regional and Functional Bureaus',
  },
  [HAND_SHAKE_ACCEPTED_PROP]: {
    text: 'Handshakes are reviewed and registered by CDOs.',
    link: "Learn more about the CDO's role.",
    term: 'CDO - Career Development Officer',
  },
  [PRE_PANEL_PROP]: {
    text: 'CDOs represent clients at the HR/CDA Assignments Panel.',
    link: "Learn more about the CDO's role.",
    term: 'CDO - Career Development Officer',
  },
  [IN_PANEL_PROP]: {
    text: 'CDOs represent clients at the HR/CDA Assignments Panel.',
    link: "Learn more about the CDO's role.",
    term: 'CDO - Career Development Officer',
  },
  [PANEL_RESCHEDULED_PROP]: {
    text: 'CDOs represent clients at the HR/CDA Assignments Panel.',
    link: "Learn more about the CDO's role.",
    term: 'CDO - Career Development Officer',
  },
  [APPROVED_PROP]: {
    text: 'AOs may be preparing your assignment, or an HR Tech is writing your orders.',
    link: "Learn more about AO and HR Tech's roles.",
    term: 'AO - Assignments Officer',
  },
};

// Numbers on the Bid Tracker for given statuses
export const DRAFT_NUMBER = 1;
export const SUBMITTED_NUMBER = 2;
export const HAND_SHAKE_OFFERED_NUMBER = 3;
export const HAND_SHAKE_ACCEPTED_NUMBER = 4;
export const IN_PANEL_NUMBER = 5;
export const APPROVED_NUMBER = 6;

// We make all of them functions so that we don't have to guess whether to call a function or not
export const GET_DRAFT_TITLE = () => ['Draft'];
export const GET_SUBMIT_BID_ACTION_TITLE = () => ['Submit Bid'];
export const GET_SUBMIT_BID_COMPLETE_TITLE = () => ['Bid Submitted'];
export const GET_HAND_SHAKE_EVALUATE_TITLE = () => ['Evaluate Bid'];
export const GET_HAND_SHAKE_EVALUATING_TITLE = () => ['Pending Review'];
export const GET_BID_REVIEW_COMPLETE_TITLE = () => ['Bid Reviewed'];
export const GET_HAND_SHAKE_OFFERED_TITLE = () => ['Handshake Offered'];
export const GET_HAND_SHAKE_ACCEPTED_TITLE = () => ['Handshake Accepted'];
export const GET_HAND_SHAKE_NEEDS_REGISTER_TITLE = () => ['Register Handshake'];
export const GET_HAND_SHAKE_COMPLETE_REGISTER_TITLE = () => ['Handshake Registered'];
export const GET_PANEL_TITLE = () => ['Panel Scheduled'];
export const GET_APPROVAL_TITLE = () => ['Assignment Approved'];

// all the statuses considered 'submitted' and still 'active' on the bid tracker
export const BID_TRACKER_SUBMITTED_ACTIVE_STATUSES = [
  APPROVED_PROP, HAND_SHAKE_ACCEPTED_PROP, HAND_SHAKE_OFFERED_PROP,
  PRE_PANEL_PROP, IN_PANEL_PROP, SUBMITTED_PROP, PANEL_RESCHEDULED_PROP,
  HAND_SHAKE_NEEDS_REGISTER_PROP];

export const BID_TRACKER_TOOLTIP_TEXT = {
  [DRAFT_PROP]: {
    title: `What is a "${GET_DRAFT_TITLE()}"?`,
    text: `"${GET_DRAFT_TITLE()}s" are bids you have on your bid tracker but you haven’t submitted them for consideration.`,
  },
  [SUBMITTED_PROP]: {
    title: `What is "${GET_SUBMIT_BID_COMPLETE_TITLE()}"?`,
    text: `"${GET_SUBMIT_BID_COMPLETE_TITLE()}" are those bids you have researched and submitted for consideration. Once a bid cycle is “open”, Bureaus can view who bid on their positions.`,
  },
  [HAND_SHAKE_OFFERED_PROP]: {
    title: `What is "${GET_HAND_SHAKE_EVALUATING_TITLE()}"?`,
    text: `"${GET_HAND_SHAKE_EVALUATING_TITLE()}" means your bid is currently being reviewed by the bureau for consideration (the Bureau confers with the Post or with the functional bureau for a consultative staffing position). Once you are offered and accept a handshake the CDO will review to ensure your handshake is ready to be registered. A CDO verifies if a bidder is an eligible bidder before the handshake can be registered.`,
  },
  [HAND_SHAKE_ACCEPTED_PROP]: {
    title: `What is "${GET_HAND_SHAKE_COMPLETE_REGISTER_TITLE()}"?`,
    text: `"${GET_HAND_SHAKE_COMPLETE_REGISTER_TITLE()}" reflects the date your handshake was registered by the CDO. Next the GTM/CDA Assignment Officer for your onward post prepares your assignment for panel.`,
  },
  [PRE_PANEL_PROP]: {
    title: `What is "${GET_PANEL_TITLE()}"?`,
    text: `"${GET_PANEL_TITLE()}" displays the date you are scheduled to be paneled once your position has been added to the panel agenda.`,
  },
  [APPROVED_PROP]: {
    title: `What is "${GET_APPROVAL_TITLE()}"?`,
    text: `"${GET_APPROVAL_TITLE()}" is the date your assignment was officially approved at panel (Hover over this status for a surprise!). Within a few days of being approved at panel, you will receive your Travel Message (TMOne) announcing to your current and future post your approved assignment.`,
  },
};
