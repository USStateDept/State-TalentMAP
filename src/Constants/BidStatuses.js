import {
  APPROVED_PROP,
  CLOSED_PROP,
  DECLINED_PROP,
  DRAFT_PROP,
  HAND_SHAKE_ACCEPTED_PROP,
  HAND_SHAKE_DECLINED_PROP,
  HAND_SHAKE_NEEDS_REGISTER_PROP,
  HAND_SHAKE_OFFERED_PROP,
  IN_PANEL_PROP,
  PRE_PANEL_PROP,
  SUBMITTED_PROP,
} from './BidData';

// store any bid statuses from the API here
// class_name is used to pass to parent container for conditional css class name
export const APPROVED = { property: APPROVED_PROP, class_name: 'approved', text: 'Approved' };
export const CLOSED = { property: CLOSED_PROP, class_name: 'closed', text: 'Closed' };
export const DRAFT = { property: DRAFT_PROP, class_name: 'draft', text: 'Draft bid' };
export const DECLINED = { property: DECLINED_PROP, class_name: 'declined', text: 'Bid unapproved' };
export const HAND_SHAKE_ACCEPTED = { property: HAND_SHAKE_ACCEPTED_PROP, class_name: 'hand-shake-accepted', text: 'Handshake registered' };
export const HAND_SHAKE_OFFERED = { property: HAND_SHAKE_OFFERED_PROP, class_name: 'hand-shake-offered', text: 'Handshake offered' };
// TODO ensure HAND_SHAKE_DECLINED has accurate data
export const HAND_SHAKE_DECLINED = { property: HAND_SHAKE_DECLINED_PROP, class_name: 'declined', text: 'Handshake declined' };
export const HAND_SHAKE_NEEDS_REGISTER = { property: HAND_SHAKE_NEEDS_REGISTER_PROP, class_name: 'registered', text: 'Handshake registered' };
// TODO ensure PRE-PANEL exists
export const PRE_PANEL = { property: PRE_PANEL_PROP, class_name: 'pre-panel', text: 'Pre panel' };
export const IN_PANEL = { property: IN_PANEL_PROP, class_name: 'in-panel', text: 'In panel' };
export const SUBMITTED = { property: SUBMITTED_PROP, class_name: 'submitted', text: 'Bid submitted' };

const statusArray = [
  APPROVED, CLOSED, DRAFT, DECLINED, HAND_SHAKE_ACCEPTED, PRE_PANEL,
  HAND_SHAKE_OFFERED, IN_PANEL, SUBMITTED, HAND_SHAKE_DECLINED,
];

export const BID_STATUS_ORDER = {
  [DECLINED_PROP]: 1,
  [CLOSED_PROP]: 2,
  [HAND_SHAKE_DECLINED_PROP]: 3,
  [DRAFT_PROP]: 4,
  [SUBMITTED_PROP]: 5,
  [HAND_SHAKE_OFFERED_PROP]: 6,
  [HAND_SHAKE_NEEDS_REGISTER_PROP]: 7,
  [HAND_SHAKE_ACCEPTED_PROP]: 8,
  [PRE_PANEL_PROP]: 9,
  [IN_PANEL_PROP]: 10,
  [APPROVED_PROP]: 11,
};

// find the correct object based on status, then return the class_name
export const getStatusProperty = (statusName, property = 'class_name') => {
  const matchingStatus = statusArray.find(s => s.property === statusName);
  return matchingStatus ? matchingStatus[property] : '';
};
