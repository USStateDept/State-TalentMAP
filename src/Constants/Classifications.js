import { faBalanceScaleLeft, faChessQueen, faDiceSix,
  faGraduationCap, faLanguage, faMedal, faPercentage,
  faRegistered, faSeedling, faSpa, faTree, faUserFriends } from '@fortawesome/free-solid-svg-icons';

// eslint-disable-next-line import/prefer-default-export
export const Icons = {
  3: {
    isIcon: true,
    name: faSeedling,
    text: '3rd Tour',
    shortCode: '3TR',
  },
  4: {
    isIcon: true,
    name: faTree,
    text: 'Tenured',
    shortCode: 'TEN',
  },
  R: {
    isIcon: true,
    name: faRegistered,
    text: 'Recommended for Tenure',
    shortCode: 'REC',
  },
  6: {
    isIcon: true,
    name: faDiceSix,
    text: '6/8',
    shortCode: '6/8',
  },
  A: {
    isIcon: true,
    name: faChessQueen,
    text: 'Ambassador / Deputy Assistant Secretary',
    shortCode: 'AMB',
  },
  C: {
    isIcon: true,
    name: faLanguage,
    text: 'Critical Need Language',
    shortCode: 'CNL',
  },
  C1: {
    isIcon: true,
    name: faLanguage,
    text: 'Critical Need Language 1st Tour',
    shortCode: 'CL1',
  },
  CC: {
    isIcon: true,
    name: faLanguage,
    text: 'Critical Need Language Final Tour',
    shortCode: 'CLF',
  },
  D: {
    isIcon: true,
    name: faPercentage,
    text: 'Differential',
    shortCode: 'DIF',
  },
  F: {
    isIcon: true,
    name: faBalanceScaleLeft,
    text: 'Fair Share',
    shortCode: 'FAIR',
  },
  F1: {
    isIcon: true,
    name: faGraduationCap,
    text: 'Pickering Fellows',
    shortCode: 'PIK',
  },
  F2: {
    isIcon: true,
    name: faGraduationCap,
    text: 'Rangel Fellows',
    shortCode: 'RGL',
  },
  M: {
    isIcon: true,
    name: faMedal,
    text: 'Meritorious Step Increases',
    shortCode: 'MER',
  },
  P: {
    isIcon: true,
    name: faGraduationCap,
    text: 'Pickering/Rangel Fellows',
    shortCode: 'FEL',
  },
  T: {
    isIcon: true,
    name: faUserFriends,
    text: 'Tandem',
    shortCode: 'TAN',
  },
  8: {
    isIcon: true,
    name: faSpa,
    text: '8 Rule',
    shortCode: '8 Rule',
  },
};
