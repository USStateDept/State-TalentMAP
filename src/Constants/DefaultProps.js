export { initialState as DEFAULT_HIGHLIGHT_POSITION } from '../reducers/highlightPosition';

export const ACCORDION_SELECTION = { main: '', sub: '' };

export const DEFAULT_USER_PROFILE = {
  user: {
    username: '',
    first_name: '',
    last_name: '',
  },
  display_name: '...', // show '...' when loading
  initials: '',
  favorite_positions: [],
};

export const POSITION_RESULTS_OBJECT = {
  results: [],
};

export const DEFAULT_FAVORITES_COUNTS = {
  favorites: 0,
  favoritesPV: 0,
  favoritesTandem: 0,
  favoritesPVTandem: 0,
  all: 0,
};

export const DEFAULT_FAVORITES = {
  favorites: [],
  favoritesPV: [],
  favoritesTandem: [],
  favoritesPVTandem: [],
  counts: DEFAULT_FAVORITES_COUNTS,
};

export const PROFILE_MENU_SECTION_EXPANDED_OBJECT = {
  title: '',
  display: false,
};

export default ACCORDION_SELECTION;

export const DEFAULT_HOME_PAGE_RECOMMENDED_POSITIONS = {
  positions: [],
  name: '',
};

export const DEFAULT_HOME_PAGE_FEATURED_POSITIONS = {
  positions: [],
  name: '',
};
