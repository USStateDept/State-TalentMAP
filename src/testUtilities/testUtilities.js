import sinon from 'sinon';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';
import api from '../api';

// Tests all functions within a mapDispatchToProps function.
// We take two arguments, one is the entire mapDispatchToProps function, which
// can be exported from the Container/Component. The second is a config
// object with any parameters we want to pass to the function with the same name.
// For example, to pass the argument "test" to the getData() function, and "test"
// and {1} to submitData() function, we could create an object such as
// { getData: ["test"], submitData: ["test", 1] }
// Then we spy dispatch and call each property of the mapDispatchToProps
// object. Calling each property should trigger the call count on the spy,
// which we test. We reset call count between tests since we're using the
// calledOnce method.
export function testDispatchFunctions(mapDispatchToProps, config = {}) {
  it('should call all mapDispatchToProps actions', () => {
    // create spy
    const dispatchSpy = sinon.spy();
    // object with all functions
    const spiedProps = mapDispatchToProps(dispatchSpy);
    // iterate on each key
    Object.keys(spiedProps).forEach((prop) => {
      // set the prop to test
      const propToTest = spiedProps[prop];
      // set any params passed in the config object
      let params = [];
      // if there any arguments for that param, set them
      if (config[prop]) { params = config[prop]; }
      // call the prop
      propToTest(...params);
      // ensure it was called once
      sinon.assert.calledOnce(dispatchSpy);
      // reset so that called count is 0
      dispatchSpy.reset();
    });
  });
}

export function setupAsyncMocks() {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const mockAdapter = new MockAdapter(api());

  return { mockStore, mockAdapter };
}

// Form a mock adapter using a url, response, type, and callbackType.
// Spy on the mock and return the mock and the spy.
export const spyMockAdapter = ({ url = '', response = [], type = 'onGet', callbackType = 'reply' }) => {
  const mockAdapter = new MockAdapter(api());
  const spy = sinon.spy(() => response);
  const mock = () => mockAdapter[type](url)[callbackType](spy);
  return { mock, spy };
};

// Expect that some assertion was performed on a provided spy.
// Optional callback once complete.
export const expectMockWasCalled = ({ spy = () => {}, cb = () => {}, timeout = 0, assertion = 'calledOnce' }) => {
  setTimeout(() => {
    sinon.assert[assertion](spy);
    cb();
  }, timeout);
};
