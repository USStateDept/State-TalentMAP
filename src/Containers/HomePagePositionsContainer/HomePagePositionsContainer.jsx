import { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { homePageFeaturedPositionsFetchData, homePageRecommendedPositionsFetchData } from 'actions/homePagePositions';
import { BID_RESULTS, EMPTY_FUNCTION, USER_PROFILE } from 'Constants/PropTypes';
import { DEFAULT_HOME_PAGE_FEATURED_POSITIONS, DEFAULT_HOME_PAGE_RECOMMENDED_POSITIONS } from 'Constants/DefaultProps';
import HomePagePositions from '../../Components/HomePagePositions/HomePagePositions';
import Spinner from '../../Components/Spinner';

const HomePagePositionsContainer = props => {
  useEffect(() => {
    if (props.userProfile.id) {
      props.homePageRecommendedPositionsFetchData(props.userProfile.employee_info.skills,
        props.userProfile.employee_info.grade, props.userProfile.employee_info.skills_additional);
      props.homePageFeaturedPositionsFetchData(props.userProfile.employee_info.skills,
        props.userProfile.employee_info.grade, props.userProfile.employee_info.skills_additional);
    }
  }, []);

  const isFirstRun = useRef(true);
  useEffect(() => {
    if (isFirstRun.current) {
      isFirstRun.current = false;
      return;
    }
    if (props.userProfile.id) {
      props.homePageRecommendedPositionsFetchData(props.userProfile.employee_info.skills,
        props.userProfile.employee_info.grade, props.userProfile.employee_info.skills_additional);
      props.homePageFeaturedPositionsFetchData(props.userProfile.employee_info.skills,
        props.userProfile.employee_info.grade, props.userProfile.employee_info.skills_additional);
    }
  }, [props.userProfile.id]);

  const { homePageRecommendedPositions, homePageFeaturedPositions,
    userProfileIsLoading, homePageRecommendedPositionsIsLoading,
    homePageFeaturedPositionsIsLoading, bidList, userProfile } = props;

  return (
    <div className="content-container">
      {
        (userProfileIsLoading || homePageRecommendedPositionsIsLoading
              || homePageFeaturedPositionsIsLoading)
          ?
          <div className="usa-grid-full homepage-positions-section-container">
            <Spinner type="homepage-position-results" size="big" />
          </div>
          :
          <HomePagePositions
            homePageRecommendedPositions={homePageRecommendedPositions}
            homePageRecommendedPositionsIsLoading={homePageRecommendedPositionsIsLoading}
            homePageFeaturedPositions={homePageFeaturedPositions}
            homePageFeaturedPositionsIsLoading={homePageFeaturedPositionsIsLoading}
            userProfile={userProfile}
            bidList={bidList}
          />
      }
    </div>
  );
};

HomePagePositionsContainer.propTypes = {
  userProfile: USER_PROFILE.isRequired,
  bidList: BID_RESULTS.isRequired,
  userProfileIsLoading: PropTypes.bool,
  homePageRecommendedPositionsFetchData: PropTypes.func,
  homePageRecommendedPositions: PropTypes.shape({}),
  homePageRecommendedPositionsIsLoading: PropTypes.bool,
  homePageFeaturedPositionsFetchData: PropTypes.func,
  homePageFeaturedPositions: PropTypes.shape({}),
  homePageFeaturedPositionsIsLoading: PropTypes.bool,
};

HomePagePositionsContainer.defaultProps = {
  userProfile: {},
  userProfileIsLoading: false,
  homePageFeaturedPositionsFetchData: EMPTY_FUNCTION,
  homePageFeaturedPositions: DEFAULT_HOME_PAGE_FEATURED_POSITIONS,
  homePageFeaturedPositionsIsLoading: false,
  homePageRecommendedPositionsFetchData: EMPTY_FUNCTION,
  homePageRecommendedPositions: DEFAULT_HOME_PAGE_RECOMMENDED_POSITIONS,
  homePageRecommendedPositionsIsLoading: false,
};

const mapStateToProps = state => ({
  userProfile: state.userProfile,
  userProfileIsLoading: state.userProfileIsLoading,
  homePageFeaturedPositions: state.homePageFeaturedPositions,
  homePageFeaturedPositionsIsLoading: state.homePageFeaturedPositionsIsLoading,
  homePageRecommendedPositions: state.homePageRecommendedPositions,
  homePageRecommendedPositionsIsLoading: state.homePageRecommendedPositionsIsLoading,
});

export const mapDispatchToProps = dispatch => ({
  homePageFeaturedPositionsFetchData: (skills, grade, cones) =>
    dispatch(homePageFeaturedPositionsFetchData(skills, grade, cones)),
  homePageRecommendedPositionsFetchData: (skills, grade, cones) =>
    dispatch(homePageRecommendedPositionsFetchData(skills, grade, cones)),
});

export default connect(mapStateToProps, mapDispatchToProps)(HomePagePositionsContainer);
