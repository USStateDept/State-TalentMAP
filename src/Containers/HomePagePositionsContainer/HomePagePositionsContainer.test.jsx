import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import TestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { DEFAULT_HOME_PAGE_FEATURED_POSITIONS, DEFAULT_HOME_PAGE_RECOMMENDED_POSITIONS } from 'Constants/DefaultProps';
import { testDispatchFunctions } from '../../testUtilities/testUtilities';
import HomePagePositionsContainer, { mapDispatchToProps } from './HomePagePositionsContainer';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Home', () => {
  const props = {
    userProfile: {
      id: 4,
      employee_info: {
        grade: '03',
        skills: [{
          code: '0030',
          description: 'Computer Science',
        }],
      },
    },
    bidList: [],
    homePageFeaturedPositions: DEFAULT_HOME_PAGE_FEATURED_POSITIONS,
    homePageRecommendedPositions: DEFAULT_HOME_PAGE_RECOMMENDED_POSITIONS,
    homePageFeaturedPositionsFetchData: () => {},
  };
  it('is defined', () => {
    const wrapper = TestUtils.renderIntoDocument(<Provider store={mockStore({})}><MemoryRouter>
      <HomePagePositionsContainer
        {...props}
      />
    </MemoryRouter></Provider>);
    expect(wrapper).toBeDefined();
  });
  // hook-based component. going to wait to use another test framework
  xit('calls homePageFeaturedPositionsFetchData', () => {
    const spy = sinon.spy();
    const wrapper = shallow(
      <HomePagePositionsContainer.WrappedComponent
        {...props}
        homePageFeaturedPositionsFetchData={spy}
      />);
    expect(wrapper).toBeDefined();
    wrapper.instance().homePageFeaturedPositions();
    sinon.assert.calledOnce(spy);
  });

  it('matches snapshot', () => {
    const wrapper = TestUtils.renderIntoDocument(<Provider store={mockStore({})}><MemoryRouter>
      <HomePagePositionsContainer
        {...props}
      />
    </MemoryRouter></Provider>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  const config = {
    homePageFeaturedPositionsFetchData: [['1'], ['2']],
    homePageRecommendedPositionsFetchData: [['1'], ['2']],
  };
  testDispatchFunctions(mapDispatchToProps, config);
});
