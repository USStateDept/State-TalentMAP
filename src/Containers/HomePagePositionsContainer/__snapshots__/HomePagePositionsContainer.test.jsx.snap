// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `undefined`;
